packages/cypress/src/add-commands/login.ts

```diff
@@ -125,26 +122,146 @@ function loginByForm(commandOptions: CommandLoginOptions) {
       userCredentials.email,
       commandOptions.entryPointUriPath,
     ];
-    // const mcUrl = new URL(appConfig.mcApiUrl);
-    // const mcFrontendHostname = mcUrl.hostname.replace('mc-api', 'mc');
 
+    /**
+     * The function is used by Cypress `session` command to store the
+     * browser state after executing the authentication flow.
+     */
     function authCallback() {
-      cy.visit(url, { onBeforeLoad: commandOptions.onBeforeLoad });
-
-      // NOTE: using `cy.origin` is currently disabled as it does not seem to properly work.
-      // Interestingly, starting an application locally using Vite works, however not when using Webpack.
-      // For now we keep it disabled until we find a solution.
-
-      // // https://cypress.io/blog/2022/04/25/cypress-9-6-0-easily-test-multi-domain-workflows-with-cy-origin/
-      // cy.origin(
-      //   mcFrontendHostname,
-      //   { args: userCredentials },
-      //   fillLoginForm
-      // );
-      fillLoginForm(userCredentials);
-
-      // Wait for the route to be loaded so that the session can be saved.
-      cy.url().should('include', url);
+      // Get the feature flags from the Merchant Center API
+      cy.request({
+        method: 'POST',
+        url: `${appConfig.mcApiUrl}/graphql`,
+        body: {
+          operationName: 'AllFeatures',
+          query: `query AllFeatures { allFeatures { name value reason } }`,
```

Comment:
We're fetching the feature flags from the MC API to see if Identity is enabled. This should make it easier to roll out.